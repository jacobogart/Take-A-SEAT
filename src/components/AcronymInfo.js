import React, { Component } from 'react';

class AcronymInfo extends Component {
  render() {
    return (
      <div className="letterHolder">
        <div>
          {this.props.stageName
            .split("")
            .map((letter, index) => 
              <h3 
                className="restOfWord"
                key={index}
              >
                {letter}
              </h3>)
            .slice(1)}
        </div>
        <p className="stageDescription">{this.props.stageDescription}</p>
      </div>
    );
  }
}

export default AcronymInfo;
