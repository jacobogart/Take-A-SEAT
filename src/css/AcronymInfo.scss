.AcronymInfo {
  position: absolute;
  z-index: 10;
  height: 500%;
  display: flex;
}


.restOfWord {
  font-size: 30px;
  margin: 0;
}

.stageDescription {
  font-size: 20px;
  width: 200px;
}