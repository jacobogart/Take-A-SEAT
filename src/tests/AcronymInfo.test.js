import React from 'react';
import AcronymInfo from '../components/AcronymInfo';
import { shallow } from 'enzyme';

const mockName = "Setup";
const mockDescription =
  "Setup the conditions required to execute the action on your Subject Under Test";

describe('AcronymInfo', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <AcronymInfo 
        stageName={mockName}
        stageDescription={mockDescription}
      />)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});