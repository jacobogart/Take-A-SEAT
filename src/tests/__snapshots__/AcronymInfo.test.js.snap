// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcronymInfo should match snapshot 1`] = `
<div
  className="letterHolder"
>
  <div>
    <h3
      className="restOfWord"
      key="1"
    >
      e
    </h3>
    <h3
      className="restOfWord"
      key="2"
    >
      t
    </h3>
    <h3
      className="restOfWord"
      key="3"
    >
      u
    </h3>
    <h3
      className="restOfWord"
      key="4"
    >
      p
    </h3>
  </div>
  <p
    className="stageDescription"
  >
    Setup the conditions required to execute the action on your Subject Under Test
  </p>
</div>
`;
